File: src/js/stringRemoveDuplicates.js
/**
 * @param {string} S
 * @return {string}
 */
var removeDuplicates = function(s) {
  let first = 0;
  let second = 1;
  
  while (s[second]) {
      if (s[first] === s[second]) {
          s = s.slice(0, first) + s.slice(second + 1);
          first = 0;
          second = 1;
      } else {
          first += 1;
          second += 1;
      }
  }
  
  return s
};

/**
 * @param {string} S
 * @return {string}
 */
var removeDuplicates = function(s) {
    let pointer = 0
    
    while (s[pointer]) {
        if (s[pointer] === s[pointer + 1]) {
            s = s.slice(0, pointer) + s.slice(pointer + 2);
            pointer = 0;
        } else {
            pointer++;
        }
    }
    
    return s
};

File: src/go/stringRemoveDuplicates.go
func removeDuplicates(S string) string {
	i := 0
	
	for i < len(S) - 1 {
		if S[i] == S[i + 1] {
			S = S[0: i] + S[i + 2:]
			i = 0
		} else {
			i += 1
		}
	}

	return S
}